src/java.base/share/classes/java/lang/AssertionError.java

```diff
@@ -71,6 +71,7 @@ private AssertionError(String detailMessage) {
      * @param detailMessage value to be used in constructing detail message
      * @see   Throwable#getCause()
      */
+    @SuppressWarnings("this-escape")
     public AssertionError(Object detailMessage) {
```

Comment:
The Javadoc of this constructor should probably mention that it calls `initCause(…)` when `detailMessage` is a `Throwable`.